cloudflare/resource_cloudflare_access_application_test.go

```diff
@@ -497,31 +542,27 @@ func testAccCheckCloudflareAccessApplicationDestroy(s *terraform.State) error {
 			continue
 		}
 
-		_, err := client.AccessApplication(context.Background(), rs.Primary.Attributes["zone_id"], rs.Primary.ID)
-		if err == nil {
-			return fmt.Errorf("AccessApplication still exists")
+		var notFoundError *cloudflare.NotFoundError
+		if rs.Primary.Attributes["zone_id"] != "" {
+			_, err := client.ZoneLevelAccessApplication(context.Background(), rs.Primary.Attributes["zone_id"], rs.Primary.ID)
```

Comment:
this addressed a long standing silent bug hidden in the depths of the suite. we were previously only ever checking account level applications so it was always passing.